### PR TITLE
Refactor addcommand

### LINKED ISSUE #44: As a reader, I want to add a new book for it to be tracked

### PR DESCRIPTION
**Pull request by pennhanlee**
Closes #44 

The Add book command is refactored to now accept books instead of person. 
eg. `add n/Naruto g/Manga total/500 bookmark/1 t/Anime t/Japanese`

Code
* Refactored Add command to take in Total Pages and Book Mark|
* Refactored Error Messages
* Bookmark is optional and will be defaulted to 0 if not indicated 

Test
* Updated tests under Add Command and its related unit tests
* Enabled all previously disabled tests related to Add Command
* Created several new constants for Testing purposes

### CODE REVIEW

File: src/main/java/seedu/bookmark/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
     public static final Prefix PREFIX_NAME = new Prefix("n/");
     public static final Prefix PREFIX_GENRE = new Prefix("g/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
-    public static final Prefix PREFIX_TOTAL_PAGES = new Prefix("tp/");
-    public static final Prefix PREFIX_BOOKMARK = new Prefix("b/");
+    public static final Prefix PREFIX_TOTAL_PAGES = new Prefix("total/");
+    public static final Prefix PREFIX_BOOKMARK = new Prefix("bookmark/");
```
Comment (TanLeYang) [inline src/main/java/seedu/bookmark/logic/parser/CliSyntax.java]: According to the user guide, the prefix should be "tp/" for total pages and "b/" for bookmark for consistency purposes right? Maybe we can discuss this further!

File: src/test/java/seedu/bookmark/logic/commands/AddCommandTest.java [hunk 2]
```diff
@@ -170,19 +170,19 @@ public boolean hasBook(Book book) {
     /**
      * A Model stub that always accept the person being added.
```
Comment (TanLeYang) [inline src/test/java/seedu/bookmark/logic/commands/AddCommandTest.java]: Thanks for doing the renaming! There's some parts which haven't been renamed such as the Javadoc here, comments in the `equals()` test as well as here: https://github.com/AY2021S1-CS2103T-F13-2/tp/blob/c568a3c5177c68a33c3e6279923fa8f3c0356df0/src/test/java/seedu/bookmark/logic/commands/AddCommandTest.java#L153
Would be great if you could change those too, thanks!

File: src/main/java/seedu/bookmark/logic/parser/CliSyntax.java (hunk 1 above)
Comment (pangpuncake) [reply src/main/java/seedu/bookmark/logic/parser/CliSyntax.java]: The rationale we discussed was for it to be quicker to type for fast typers!

File: src/main/java/seedu/bookmark/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -89,4 +92,32 @@ public static Tag parseTag(String tag) throws ParseException {
...
+            throw new ParseException(TotalPages.MESSAGE_CONSTRAINTS);
+        }
+        return new TotalPages(trimmedTotalPages);
+    }
+
+    /**
+     * Parses {@code Optional<String> bookmark} into a {@code Bookmark}
+     */
+    public static Bookmark parseBookmark(Optional<String> bookmark, TotalPages totalPages) throws ParseException {
+        if (bookmark.equals(Optional.empty())) {
```
Comment (pangpuncake) [inline src/main/java/seedu/bookmark/logic/parser/ParserUtil.java]: Perhaps using `bookmark.isEmpty()` could replace `bookmark.equals(Optional.empty())`

**Review by pangpuncake — CHANGES_REQUESTED** [review]
Few fixes, other than that, LGTM!

File: src/main/java/seedu/bookmark/logic/parser/CliSyntax.java (hunk 1 above)
Comment (pennhanlee) [reply src/main/java/seedu/bookmark/logic/parser/CliSyntax.java]: Looks like there are some discrepancies between our UG and DG. Let's go over this again in the next meeting. In the meantime, I'll make the relevant changes! Thanks!

File: src/test/java/seedu/bookmark/logic/commands/AddCommandTest.java (hunk 2 above)
Comment (pennhanlee) [reply src/test/java/seedu/bookmark/logic/commands/AddCommandTest.java]: Must have missed this part out. Thanks!

File: src/main/java/seedu/bookmark/logic/parser/ParserUtil.java (hunk 3 above)
Comment (pennhanlee) [reply src/main/java/seedu/bookmark/logic/parser/ParserUtil.java]: Good recommendation! Thank you for spotting this. I've made the relevant changes.

**Review by TanLeYang — APPROVED** [review]
LGTM!

**Review by pangpuncake — APPROVED** [review]
LGTM!

**Pull request merged**